pkg/reconciler/openshift/tektonaddon/openshiftConsole.go

```diff
@@ -25,13 +25,54 @@ import (
 	"github.com/tektoncd/operator/pkg/apis/operator/v1alpha1"
 	"github.com/tektoncd/operator/pkg/reconciler/common"
 	"github.com/tektoncd/operator/pkg/reconciler/kubernetes/tektoninstallerset/client"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func (r *Reconciler) EnsureOpenShiftConsoleResources(ctx context.Context, ta *v1alpha1.TektonAddon) error {
-	if err := r.installerSetClient.CustomSet(ctx, ta, OpenShiftConsoleInstallerSet, r.openShiftConsoleManifest, filterAndTransformOCPResources()); err != nil {
-		return err
+func (r *Reconciler) EnsureOpenShiftConsoleResources(ctx context.Context, ta *v1alpha1.TektonAddon) (error, bool) {
+	var filteredManifest mf.Manifest
+	consoleYamlSampleExist, err := r.checkCRDExist(ctx, "consoleyamlsamples.console.openshift.io")
```

Comment:
should we have this check while starting the controller?
if controller detects CRD it adds a boolean in `Reconciler` and  latrer it creates resources based on that otherwise not
we don't probably need to check in every reconciliation?

Reply:
I did it to install the resources if the devconsole operator is installed at later stage.

Reply:
@piyush-garg when would this change be picked up ? If the console is not available when we install the operator, but then becomes available, when will the operator pick up that change ? My guess would be during the "refresh" reconcile loop, but I wondered 👼🏼.